bundles/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/PLCLogoBinding.java

```diff
@@ -0,0 +1,562 @@
+                    } else {
+                        logger.info("Found unknown model for {}: {}", controllerName, modelName);
+                    }
+
+                    if (model != null) {
+                        deviceConfig.setModel(model);
+                        logger.info("Set model for {}: {}", controllerName, modelName);
+                    }
+                }
+            } // while
+
+            Iterator<Entry<String, PLCLogoConfig>> entries = controllers.entrySet().iterator();
+            while (entries.hasNext()) {
+                Entry<String, PLCLogoConfig> thisEntry = entries.next();
+                String controllerName = thisEntry.getKey();
+                PLCLogoConfig deviceConfig = thisEntry.getValue();
+                S7Client LogoS7Client = deviceConfig.getS7Client();
+                if (LogoS7Client == null) {
+                    LogoS7Client = new Moka7.S7Client();
+                } else {
+                    LogoS7Client.Disconnect();
+                }
+                LogoS7Client.SetConnectionParams(deviceConfig.getlogoIP(), deviceConfig.getlocalTSAP(),
+                        deviceConfig.getremoteTSAP());
+                logger.info("About to connect to {}", controllerName);
+
+                if ((LogoS7Client.Connect() == 0) && LogoS7Client.Connected) {
+                    logger.info("Connected to PLC LOGO! device {}", controllerName);
+                } else {
+                    logger.error("Could not connect to PLC LOGO! device {} : {}", controllerName,
+                            S7Client.ErrorText(LogoS7Client.LastError));
+                    throw new ConfigurationException("Could not connect to PLC LOGO! device ",
+                            controllerName + " " + deviceConfig.getlogoIP());
+                }
+                deviceConfig.setS7Client(LogoS7Client);
+            }
+
+            setProperlyConfigured(configured);
+        } else {
+            logger.info("No configuration for PLCLogoBinding");
+        }
+    }
+
+    private State createState(Item item, Object value) {
+        DecimalType number = null;
+        if (value instanceof Number) {
+            number = new DecimalType(value.toString());
+        }
+
+        State state = null;
+        if (item instanceof StringType) {
+            state = new StringType((String) value);
+        } else if (item instanceof NumberItem) {
+            if (number != null) {
+                return number;
+            } else if (value instanceof String) {
+                state = new DecimalType(((String) value).replaceAll("[^\\d|.]", ""));
+            }
+        } else if (item instanceof SwitchItem && (number != null)) {
+            state = (number.intValue() > 0) ? OnOffType.ON : OnOffType.OFF;
+        } else if (item instanceof ContactItem && (number != null)) {
+            state = (number.intValue() > 0) ? OpenClosedType.CLOSED : OpenClosedType.OPEN;
+        }
+        return state;
+    }
+
```

Comment:
In order for the binding to work on openHAB 2, you might need to add these two methods:

```
    protected void addBindingProvider(PlcLogoBindingProvider bindingProvider) {
        super.addBindingProvider(bindingProvider);
    }

    protected void removeBindingProvider(PlcLogoBindingProvider bindingProvider) {
        super.removeBindingProvider(bindingProvider);
    }

```

If you haven't already, please download a recent openHAB 2 snapshot build and verify that the JAR works there.